packages/core/contracts/oracle/implementation/Staker.sol

```diff
@@ -113,14 +113,19 @@ abstract contract Staker is StakerInterface, Ownable, Lockable {
      * be added to the pending stake. If not, the stake amount will be added to the active stake.
      * @param amount the amount of tokens to stake.
      */
-    function stake(uint256 amount) public override nonReentrant() {
-        VoterStake storage voterStake = voterStakes[msg.sender];
+    function stake(uint256 amount) public {
+        _stake(msg.sender, amount);
+    }
+
+    function _stake(address wallet, uint256 amount) internal {
```

Comment:
so the point of changing this is to let the delegate _stake_ to the address that delegated to them (the delgator)?

a few points:
1) I think it would be better to do both this change (and the change to `withdrawRewards`) without making the internal functions. It seems a bit unnesasary as when you call the internal function you are simply passing in `msg.sender` in both cases.
2) do you not need to relate the `msg.sender` back to the delegate using the `getVoterFromDelegate` method? (this method is currently [defined](https://github.com/UMAprotocol/protocol/blob/855e3c740938f8c6a5675387225d3e059b6c1d70/packages/core/contracts/oracle/implementation/VotingV2.sol#L650) in the `VotingV2.sol` contract but we could easily move it here to re-use the logic within the staking context) 
3) (bonus) It would be pretty cool if you could call a new method `stakeTo` as any wallet where it'll pull funds from your EOA and stake them in a provided accounts balance.

Reply:
on both 1 and 2 I see now that you have `withdrawAndRestake` within the voting contract! makes more sense now :)

I think my points still hold: a) remove the internal methods, if you can and b) try get this logic into this `Staker` contract by pulling some of the logic you need into this implementation (in particular the `getVoterFromDelegate` method can be defined here).

Reply:
Thanks for your review @chrismaree!

a) I've removed all the internal functions I added except for `_stakeTo` now that we have a public `stakeTo` so it can be reused by the regular `stake`
b) I've moved all the logic to `Staker`, I agree it makes more sense

Since I now  we need to call `getVoterFromDelegate` from `Staker`, I thought it made sense to move that function to `Staker`, as well as `setDelegate` and `setDelegator`, since the delegation state lives in `Staker`. Does it make sense?